servers/src/main/java/tachyon/worker/file/FileWorkerMasterSyncExecutor.java

```diff
@@ -79,11 +79,10 @@ public void heartbeat() {
 
     FileSystemCommand command = null;
     try {
-      command = mMasterClient.heartbeat(WorkerIdRegistry.getWorkerId(),
-          persistedFiles);
+      command = mMasterClient.heartbeat(WorkerIdRegistry.getWorkerId(), persistedFiles);
```

Comment:
If the heartbeat fails with the master, than means the state of the persisted files on this worker is now empty(), but the master doesn't know it. Could this get into a state where the master will always think the worker is persisting, but the worker has already persisted it?

It seems like the command to persist is idempotent, but the notification back to the master is not?